spec/concurrent/atomic/cyclic_barrier_spec.rb

```diff
@@ -71,6 +71,7 @@ module Concurrent
         end
 
         start_latch.wait(1)
+        Thread.pass until barrier.number_waiting == 1
```

Comment:
This might benefit from a timeout. Maybe a better way than this:

```suggestion
        start = Time.now
        Thread.pass until barrier.number_waiting == 1 || (Time.now - start > 10)
```

Also, that start_latch above maybe becomes unnecessary entirely now.

Reply:
btw, these are the test helpers I use to wait until expected conditions are met that are pretty handy: https://github.com/bensheldon/good_job/blob/9729c3a4e7339fc25044490576fbbc0136f6a45d/spec/support/sleep_helper.rb#L6-L22

Reply:
Fixed with `repeat_until_success`, thanks for the suggestion!